packages/libs/hd-wallet/src/utils/encrypt.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  createCipheriv,
+  createDecipheriv,
+  pbkdf2Sync,
+  randomBytes,
+} from 'crypto';
+
+type BinaryLike = string | NodeJS.ArrayBufferView;
+
+const SALT = Buffer.from('FkM5B23nB6LVY7mrwDeh'); // SALT value used for deriving keys
+/**
+ * Derive a cryptographic key from the provided password.
+ * @param {string} password - User's password.
+ * @returns {Buffer} - Returns the derived cryptographic key.
+ */
+function deriveKey(password: string, salt: BinaryLike): Buffer {
+  return pbkdf2Sync(password, salt, 1000, 32, 'sha256');
+}
+
+/**
+ * Encrypt the provided text using AES-256-GCM algorithm.
+ * @param {Buffer} text - Text to encrypt.
+ * @param {string} password - User's password.
+ * @returns {{ cipherText: Buffer; iv: Buffer; tag: Buffer }} - Returns the encrypted text, initialization vector, and authentication tag.
+ */
```

Comment:
I don't think it's needed to pass the type in tsdoc as it's extracted from the type of the function.

```suggestion
/**
 * Encrypt the provided text using AES-256-GCM algorithm.
 * @param text - Text to encrypt.
 * @param password - User's password.
 * @returns - Returns the encrypted text, initialization vector, and authentication tag.
 */
```